src/state.rs

```diff
@@ -550,6 +551,16 @@ impl Heap {
         result
     }
 
+    pub fn read_unaligned_from_pointer(&self, pointer: &FatPointer) -> Vec<u8> {
+        let mut result = Vec::new();
+        let start = pointer.start + pointer.offset;
+        let finish = start + pointer.len;
+        for i in start..finish {
+            result.push(self.heap[i as usize]);
```

Comment:
This is assuming the heap is always sized properly, right? This will panic if we're out of bounds, let's check that the bounds are correct or either return an error ir self.heap.get(i) is None.

Reply:
Done

Reply:
I believe the intended behaviour is to break the loop instead.